### ISSUE TITLE
Expander: non deterministic expansion on some circular ref patterns

### ISSUE THREAD

**Issue by fredbi** [go-openapi/spec#93]
Let's make things official about this limitation, mentioned in the discussion following #75 (credits @elakito):
- when $ref do form several cycles  (i.e. out of the root document)
- when some of those cycles have an intersection

then the expander chose some random expansion path from the common cycle

This behavior is illustrated by this [fixture](fixtures/expansion/circular-minimal.json) (not run by tests).

**Comment by magodo** [comment]
@fredbi Is there any progress on this issue?

**Comment by fredbi** [comment]
@magodo no. I've worked a bit on it last year then gave up since I found it very difficult to solve.
